refactor(transform/styled-components): shorten code

overlookmotel opened the pull request:
Pure refactor. Use `for` loop instead of manually updating `quasi_index`.

Convert `quasis` to a mutable slice before the loop, so hopefully compiler can see that `quasis` cannot change its length during the loop, and so no bounds checks are required on `quasis[quasi_index]` accesses.

Dunqing commented on crates/oxc_transformer/src/plugins/styled_components.rs:
```diff
@@ -894,8 +892,8 @@ fn minify_template_literal<'a>(lit: &mut TemplateLiteral<'a>, ast: AstBuilder<'a
     // TODO: What about `cooked`? Shouldn't we alter that too?
-    let mut quasi_index = 0;
-    while quasi_index < quasis.len() {
+    let quasis = &mut lit.quasis[..];
+    for quasi_index in 0..quasis.len() {
         let mut bytes = quasis[quasi_index].value.raw.as_str().as_bytes();
```
Can we use `iter().enumerate()` here?

overlookmotel replied:
Unfortunately not. Borrow-checker will not allow. The iterator would borrow `quasis`, so then you can't mutate other elements of `quasis` in the loop e.g. `quasis[quasi_index - 1].span = REMOVE_SENTINEL;`.

overlookmotel replied:
We could go unsafe code on it and get around this. But that's an optimization too far, even for me!

Dunqing replied:
Ok, I forgot that part

Dunqing reviewed (approved):
Looks good!

Dunqing commented:
### Merge activity

* **Jul 16, 12:59 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
refactor(transform/styled-components): shorten code (#12316)

Pure refactor. Use `for` loop instead of manually updating `quasi_index`.

Convert `quasis` to a mutable slice before the loop, so hopefully compiler can see that `quasis` cannot change its length during the loop, and so no bounds checks are required on `quasis[quasi_index]` accesses.
